adrs/0541531435/README.md

```diff
@@ -0,0 +1,118 @@
+multiple documentation tools?
+
+This ADR is an attend to follow the steps of other ecosystems like Elixir,
+Ruby or Rust came to a consensus about this topic.
+
+### About compilations
+
+We should tell what requires compilation in the project, which is extremely
+useful in library projects since you should be aware of such detail to consume
+the project. Or in cases that you would like to run the project in production.
+
+Having a directory that reflects such information is crucial to mitigate mistakes
+in application deployments, or misuse of libraries.
+
+### About test frameworks
+
+Sometimes you may have multiple test frameworks for your project, most likely
+for `end-to-end`, `integration`, or `unit` tests, but not limited to those
+concepts. Therefore, people have tried to come up with `./tests/integration`,
+`./tests/unit`, `./tests/end-to-end`.
+
+Although that may be a clean way to structure your tests, the problems arrive
+when you try to articulate the differences between those categories without
+getting into an in-depth conversation about X vs. Y and end up with unclear
+boundaries, relying too much on the programmer's experiences to articulate the
+most optimal definition or decision.
+
+Additionally, if you try to migrate your project to another testing tool that
+falls into one of those possible categories, you will mingle tests that are
+run differently depending on the underline tool, and that may be really
+important in some cases.
+
+### About documentation
+
+Documentation tools are amazingly diverse in the ecosystem; such tools require
+different configurations and code structure, therefore mingle tools may cause
+incompatible situations, or confusing documentation source code.
+
+Also, it is unclear today what should I do if I have multiple documentation
+websites (internal or public as an example), and how to manage them.
+
+## Resolution
+
+* You **must** follow the following directory structures:
+
+  ```text
+  .
+  ├── package.json
+  ├── bin
+  ├── dist
+  ├── docs
+  │   ├── storybook
+  │   └── website
+  │   └── website-next
+  │   └── [another website or tool]
+  ├── examples
+  ├── lib
+  ├── src
+  └── tests
+      ├── cypress
+      └── jest
+      └── puppeteer
+      └── [another testing tool]
+  ```
+
+* In the `./tests` directory, you **must** use the tool's name to reflect the
+  tool used to run the tests.
+* You **must** place `storybook` documentation under `./docs`.
+* In the `./docs` directory, you **must** separate documentation based on the
+  underline tool or context (for example, **internal** or **public**).
+* `./src` **must** required compilation step, and you **must** compile the
+  source code into `./dist` directory.
+* `./dist` **must** required to compile `./src` directory to generate
+  the source code, or any other compilation step in required cases.
+
+### About files and directories
+
+* **./package.json:** the NodeJS package configuration.
+* **./bin:** contains executables files.
```

Comment:
Might be out-of-scope of this ADR, however,  I think it's a best practice that "bin" scripts are a light wrapper around code that you'd find in src/lib

Reply:
Out-of-scope for sure